Components/DebateSignUp.js

```diff
@@ -1,37 +1,88 @@
 import React from 'react';
 import { Alert, AppRegistry, Button, StyleSheet, Text, TextInput, View } from 'react-native';
-import { StackNavigator } from 'react-navigation';
+import { DrawerNavigator } from 'react-navigation';
 import { Dropdown } from 'react-native-material-dropdown';
+import NavigationBar from 'react-native-navbar';
 
 export default class DebateSignUp extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      first_name: '',
+      last_name: '',
+      partner_preference: null,
+      debate_preference: 'debate_or_judge',
+      skill_level: 'nov',
+    };
+    this.handleDebateSignUpClick = this.handleDebateSignUpClick.bind(this);
+  }
+
   static navigationOptions = {
     drawer: () => ({
       title: 'Debate Sign Up',
     }),
   };
 
+  handleDebateSignUpClick(event) {
+    const { navigate } = this.props.navigation;
+    event.preventDefault();
+    return fetch('https://breakpath-api.herokuapp.com/sign_up_preferences', {
+      method: 'POST',
+      mode: 'no-cors',
+      body: JSON.stringify({
+        first_name: this.state.first_name,
+        last_name: this.state.last_name,
+        partner_preference: this.state.partner_preference,
+        skill_level: this.state.skill_level,
+        debate_preference: this.state.debate_preference,
+      }),
+    })
+    .then((response) => response.json())
+    .then((responseJson) => {
+      return responseJson;
+      navigate('Sorted Rooms');
```

Comment:
Just as a note, code after a return does not run. So in this scenario, it would not navigate to sorted rooms. No need to change it though, since I'll have to come back and change the handling of the response later anyways :)